src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -107,17 +110,20 @@ public Response forward(@NotNull @Valid RestAction ra) {
 
         Action message = builder.build();
 
+        final boolean validApplicationBundleEventType = isBundleApplicationEventTypeTripleValid(message.getBundle(), message.getApplication(), message.getEventType());
+        if (!validApplicationBundleEventType) {
+            return Response.status(404).build();
+        }
+
         try {
             String serializedAction = serializeAction(message);
             emitter.send(buildMessageWithId(serializedAction));
             forwardedActions.increment();
         } catch (IOException e) {
-            e.printStackTrace();  // TODO: Customise this generated block
+            e.printStackTrace();
```

Comment:
Can we replace that with a proper log entry?

Reply:
replaced :)